pkg/virtctl/version/version.go

```diff
@@ -40,21 +48,42 @@ type Version struct {
 }
 
 func (v *Version) Run() error {
-	fmt.Printf("Client Version: %s\n", fmt.Sprintf("%#v", version.Get()))
+	cmd.Printf("Client Version: %s\n", fmt.Sprintf("%#v", version.Get()))
 
-	if !clientOnly {
-		virCli, err := kubecli.GetKubevirtClientFromClientConfig(v.clientConfig)
-		if err != nil {
-			return err
-		}
+	if !clientOnly && serverVersion != nil {
+		cmd.Printf("Server Version: %s\n", fmt.Sprintf("%#v", *serverVersion))
+	}
 
-		serverInfo, err := virCli.ServerVersion().Get()
-		if err != nil {
-			return err
-		}
+	return nil
+}
 
-		fmt.Printf("Server Version: %s\n", fmt.Sprintf("%#v", *serverInfo))
+func CheckClientServerVersion(clientConfig *clientcmd.ClientConfig) {
+	clientVersion := version.Get()
+	virCli, err := kubecli.GetKubevirtClientFromClientConfig(*clientConfig)
+	if err != nil {
+		cmd.Println(err)
+		return
+	}
+	serverVersion, err = virCli.ServerVersion().Get()
+	if err != nil {
+		cmd.Println(err)
+		return
+	}
+	clientGitVersion := strings.TrimPrefix(clientVersion.GitVersion, "v")
+	serverGitVersion := strings.TrimPrefix(serverVersion.GitVersion, "v")
+	client, err := semver.Make(clientGitVersion)
+	if err != nil {
+		cmd.Println(err)
+		return
+	}
+	server, err := semver.Make(serverGitVersion)
+	if err != nil {
+		cmd.Println(err)
+		return
 	}
 
-	return nil
+	if client.Major != server.Major || client.Minor != server.Minor {
+		cmd.Printf(versionsNotAlignedWarnMessage, clientVersion, *serverVersion)
```

Comment:
I think it is not really feasible for people to maintain multiple `virtctl` commands when interacting with different kubevirt versions. I would expect that a new enough `virtctl` will not trigger warnings. Also, I would expect that we treat API changes which change behaviour of `virtctl` commands on features which are not alpha as bugs.

It may be ok to add a warning if a command fails. The rest sounds a little bit too aggressive for me.

Reply:
Pushed the requested changes